### ISSUE TITLE
Move from Containers to Folders for MBrace state etc.

### ISSUE THREAD

**Issue by isaacabraham** [mbraceproject/MBrace.Azure#57]
The newer versions of MBrace Azure are much better about data management etc. than the earlier releases. However, I would like to see the different containers move to a folder structure so that you can (optionally) specify a single container that all different runtime versions go under. Otherwise, we'll once again start with a proliferation of container hell as every time you spin up a cluster with a different version you'll implicitly create a new container for it.

The currently two structure with one "container" for runtime and one of data is fine, but I think that it should be slightly modified to have a single top-level container - probably specified by the configuration, and if not supplied then defaults to ```mbrace``` or something - under which specific versions are held.

```mbrace
   --- mbraceruntimedata000000-0.6.5a
   --- mbraceruntimedata000000-0.6.6a
   --- mbraceuserdata000000```

where under those subfolders (not containers) are the state and data etc..

The main benefit of this is that you encapsulation all the runtime data and fluff into a single container that the user does not need to know anything about, and it's easier to manage from their point of view as well.

**Comment by krontogiannis** [comment]
Cool. I will make some changes and post an update here.

**Comment by eiriktsarpalis** [comment]
I was wondering if assemblies managed by Vagabond should be placed in a container separate to the runtime state. This is since their content is assembly files, whose constitution is independent of mbrace versions and cluster state.

**Comment by isaacabraham** [comment]
That's just a new top level folder directly underneath the mbrace container.

**Comment by eiriktsarpalis** [comment]
True, my thinking was that Vagabond state is conceptually different to mbrace state, so it could be kept in a separate container altogether.

**Comment by isaacabraham** [comment]
That's true. I know though that as far as a user is concerned it's all MBrace - Vagabond is an implementation detail. You could supply a configuration property for the Vagabond container that was not set would just default to whatever the main mbrace container was?

**Comment by eiriktsarpalis** [comment]
Yeah, I guess you're right.
